stdlib/typing.pyi

```diff
@@ -661,7 +661,7 @@ class _TypedDict(Mapping[str, object], metaclass=ABCMeta):
     # can go through.
     def setdefault(self, k: NoReturn, default: object) -> object: ...
     # Mypy plugin hook for 'pop' expects that 'default' has a type variable type.
-    def pop(self, k: NoReturn, default: _T = ...) -> object: ...
+    def pop(self, k: NoReturn, default: object = ...) -> object: ...
```

Comment:
I'm not too familiar with the details of the mypy plugin mention here, but it looks like this would cause a mypy assert to fail. (In general I'm not a fan of these dummy type checker implementation detail leaky stubs)
Can we type ignore / pyright ignore this?

Reply:
Yeah, I agree that it's not a good policy to allow type checker implementation details leak into typeshed stubs. For now, we can quiet this with a "# type: ignore" comment. I can do that as part of a subsequent PR when I enable the check in pyright.